src/java/org/apache/cassandra/security/ISslContextFactory.java

```diff
@@ -0,0 +1,108 @@
+ * client_encryption_options}/{@code server_encryption_options} and provide a custom class-name implementing this
+ * interface with parameters to be used to plugin a your own way to load the SSLContext.
+ *
+ * Implementation of this interface must have a constructor with argument of type {@code Map<String,String>} to allow
+ * custom parameters, needed by the implementation, to be passed from the yaml configuration.
+ *
+ * Since on top of Netty, Cassandra is internally using JSSE SSLContext also for certain use-cases- this interface
+ * has methods for both.
+ *
+ * Below is an example of how to configure a custom implementation with parameters
+ * <pre>
+ * ssl_context_factory:
+ *       class_name: org.apache.cassandra.security.YourSslContextFactoryImpl
+ *       parameters:
+ *         key1: "value1"
+ *         key2: "value2"
+ *         key3: "value3"
+ * </pre>
+ */
+public interface ISslContextFactory
+{
+    /**
+     * Creates JSSE SSLContext.
+     *
+     * @param options EncryptionOptions that could be used for the SSL context creation
+     * @param buildTruststore {@code true} if the caller requires Truststore; {@code false} otherwise
+     * @return
+     * @throws SSLException in case the Ssl Context creation fails for some reason
+     */
+    SSLContext createJSSESslContext(EncryptionOptions options, boolean buildTruststore) throws SSLException;
+
+    /**
+     * Creates Netty's SslContext object.
+     *
+     * @param options EncryptionOptions that could be used for the SSL context creation
+     * @param buildTruststore {@code true} if the caller requires Truststore; {@code false} otherwise
+     * @param socketType {@link SocketType} for Netty's Inbound or Outbound channels
+     * @param useOpenSsl {@code true} if openSsl is enabled;{@code false} otherwise
+     * @param cipherFilter to allow Netty's cipher suite filtering, e.g.
+     * {@link io.netty.handler.ssl.SslContextBuilder#ciphers(Iterable, CipherSuiteFilter)}
+     * @return
+     * @throws SSLException in case the Ssl Context creation fails for some reason
+     */
+    SslContext createNettySslContext(EncryptionOptions options, boolean buildTruststore, SocketType socketType,
+                                     boolean useOpenSsl, CipherSuiteFilter cipherFilter) throws SSLException;
+
+    /**
+     * Initializes hot reloading of the security keys/certs. The implementation must guarantee this to be thread safe.
+     * @param options Encryption options
+     * @throws SSLException
+     */
+    void initHotReloading(EncryptionOptions options) throws SSLException;
+
+    /**
+     * Returns if any changes require the reloading of the SSL context returned by this factory.
+     * This will be called by Cassandra's periodic polling for any potential changes that will reload the SSL context
+     * . However only newer connections established after the reload will use the reloaded SSL context.
```

Comment:
Could you just move that dot at the end of the previous line please? :)

Reply:
Indeed, it looks really odd